lib/escape-url.js

```diff
@@ -0,0 +1,13 @@
+module.exports = function(url) {
+    // If url is already wrapped in quotes, remove them
+    if (/^['"].*['"]$/.test(url)) {
+        url = url.slice(1, -1);
+    }
+    // Should url be wrapped?
+    // See https://drafts.csswg.org/css-values-3/#urls
+    if (/["'() \t\n]/.test(url)) {
```

Comment:
What exactly is tested here, could you give a few examples please ?

1. `["'()=> <=\t\n]` => `\s` ?
2. `/["'() \t\n]/.test(url)` => `/["'() \t\n]/g.test(url)` (`/RegExp/g`) ?

Reply:
They are the characters outlined as requiring escaping in unquoted URLs in the CSS spec.

https://drafts.csswg.org/css-values-3/#urls

If any of them are present, the string is wrapped in quotes and quotes/newline is escaped. Like with `JSON.stringify`, yes, it's basically the same as `\s` in regex, just from different standards. There seems to be some rarely used characters regex considers whitespace that CSS doesn't, but again, it is unlikely to matter. I just prefer to be specific when the standard is right there.

Some examples that would require either escaping or wrapping:

`url(open(parens)` => `url(open\(parens)` / `url("open(parens")`

`url(close)parens)` => `url(close\)parens)` / `url("close)parens")`

`url(quote"in"middle)` => `url("quote\"in\"middle")`

Reply:
Regarding your second point: Why would it be better to test with a global regex? I'm not aware of any difference caused by this.